src/sentry/api/endpoints/project_search_details.py

```diff
@@ -119,6 +119,14 @@ def delete(self, request, project, search_id):
         except SavedSearch.DoesNotExist:
             raise ResourceDoesNotExist
 
-        search.delete()
+        is_search_owner = request.user and request.user == search.owner
 
-        return Response(status=204)
+        if request.access.has_scope('project:write'):
+            if not search.owner or is_search_owner:
```

Comment:
Doesn't this prevent admins/project:write users from deleting other people's searches now? Curious if there's a reason we need to take away those privileges - it seemed somewhat reasonable.

Reply:
It doesn't prevent them from deleting other people's shared searches (i.e. search.owner is null). It only prevents them from deleting other people's personal searches, which I don't think they can see anyway.
https://github.com/getsentry/sentry/blob/master/src/sentry/api/endpoints/project_searches.py#L35